packages/database/src/modelClasses/DataSource.js

```diff
@@ -9,9 +9,11 @@ import { TYPES } from '../types';
 
 const DATA_ELEMENT = 'dataElement';
 const DATA_GROUP = 'dataGroup';
+const SYNC_GROUP = 'syncGroup';
 const DATA_SOURCE_TYPES = {
   DATA_ELEMENT,
   DATA_GROUP,
+  SYNC_GROUP,
```

Comment:
I'm not 100% on adding this to the data source types. This means that technically the model doesn't match the db, since I haven't added `syncGroup` as a valid type in the `data_source` table. But we are conceptually using it as another data source type.

Reply:
The way I see it there are two options for what we want here, and both of them currently involve some single use special handling.

Either we're passing the `DataBroker` the data source type, i.e. `dataElement, dataGroup, syncGroup`, and there's special handling that says `syncGroup` is pulled from a separate database table. (Which is the current implementation)

I'm actually coming around on this being here, because no matter where we put the definition of a sync group type, we're still passing data broker two data sources and a pseudo data source. I think anywhere else we put it would make the code messier for no real gain.

Alternatively we could update `DataBroker` to accept the database type, i.e. `data_source, data_service_sync_group` and there's special handling that says `data_source` can be either a `dataElement` or a `dataGroup`

This would probably be better long term if we want to split `dataElement` and `dataGroup` to separate tables, but would require a lot of refactoring which I think is outside the scope of this task.

Reply:
If Ethan says something is outside the scope of the current task, I'd better listen ;-)

Happy to go with what you've got already. Long term we do want to split `dataElement` and `dataGroup`, but who knows how far away that will be.